.github/workflows/unit-tests.yml

```diff
@@ -5,45 +5,21 @@ on:
 
 jobs:
   run-tests:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-gpu-runner
     steps:
       - uses: actions/checkout@v4
 
-      - name: Install poetry
-        run: pipx install poetry
-
       - name: Set up Python version
         uses: actions/setup-python@v5
         with:
-          python-version: '3.9'
-          cache: 'poetry'
-
-      - name: qr-reader tests
-        run: |
-          make install-qr-reader
-          make test-qr-reader
+          python-version: '3.10'
 
-      - name: owlv2 tests
-        run: |
-          make install-owlv2
-          make test-owlv2
-
-      - name: zeroshot-counting tests
-        run: |
-          make install-zeroshot-counting
-          make test-zeroshot-counting
-
-      - name: depth-anything-v2 tests
-        run: |
-          make install-depth-anything-v2
-          make test-depth-anything-v2
-
-      - name: nsfw-classification tests
-        run: |
-          make install-nsfw-classification
-          make test-nsfw-classification
+      - name: Run image
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: "1.7.1"
 
-      - name: controlnet-aux tests
+      - name: tests
         run: |
-          make install-controlnet-aux
-          make test-controlnet-aux
+          make install
+          make test
```

Comment:
@camiloaz are you sure that we want to run the tests of all the models? There might be some very heavy models that might not fit on the GPU instance.

Reply:
I think for now the tests are running fine, so we should not be concerned about that. If we get to a point where the GPU does not fit the model, we can skip those specific tests. But this is much easier for everyone because you don't need to add so much boilerplate to add any tool.